src/node-red/getting-started/node-red-messages.md

```diff
@@ -0,0 +1,226 @@
+
+## What Are Node-RED Messages?
+
+In Node-RED, messages are packets of data that travel between nodes in your flow. Node-RED is event-driven, meaning an event emitter triggers events, and a listener reacts to them. In this context, nodes in Node-RED act as both event emitters and listeners, with messages serving as the medium through which these events are communicated. These messages represent the information that nodes read, modify, and act upon. Each node processes these messages and passes them along to the next node in the sequence.
+
+![Node-RED message passing animation](./images/node-red-message-passing.gif){data-zoomable}
+_Node-RED message passing animation_
+
+Messages are crucial because they carry the data that powers your workflows—whether it’s sensor readings, user inputs, or responses from an API. Essentially, Node-RED messages are JavaScript objects, which are simple yet powerful structures for managing and transferring data throughout your flows.
+
+### Understanding JavaScript Objects
+
+A [JavaScript object](https://www.youtube.com/watch?v=BRSg22VacUA) is a data structure used to store multiple values in a single variable. It consists of key-value pairs, where each key (or property) is associated with a specific value. This organization allows you to group and access related data easily.
+
+ For example: 
+
+```javascript
+{
+  name: "Bob",
+  age: 24,
+  married: true
+}
+```
+
+In this example, `name`, `age`, and `married` are properties of the object. Similarly, Node-RED messages use this structure to organize and transport data within your flows.
+
+## Anatomy of Node-RED Messages
+
+In Node-RED, messages are by default referred to as `msg`, and nodes are designed to work with this naming convention.
+
+Node-RED messages consist of several key properties essential for data handling and communication between nodes:
+
+- **`msg._msgid`**: A unique identifier for the message, automatically assigned by Node-RED. It helps in tracking and debugging messages within the flow.
+- **`msg.payload`**: This is the primary data of the message. It contains the main information that nodes process and act upon, such as sensor readings or user input. 
+- **`msg.topic`**: An optional property used to categorize or identify the message. It can be useful for routing or filtering messages based on their context.
+
+These are the most common properties you'll encounter in Node-RED, Also additional properties can be added as needed.
+
+The `_msgid` property is always present, even if you send an empty object from one node to another. This property is automatically appended by Node-RED to track messages within the flow. However, it's important to note that the `payload` and `topic` properties are not always present in every message. Their inclusion depends on whether the node appends them. Most nodes in Node-RED, including contributed nodes, use the `payload` property as a standard for communication.
```

Comment:
Maybe make a remark that properties starting with a `_` generally indicate it's not user data, but system meta data?